app/controllers/contacts_controller.rb

```diff
@@ -1,5 +1,18 @@
 class ContactsController < ApplicationController
+  before_action :authorize!
+
   def index
     @contacts = current_user.contacts
   end
+
+  private
+
+  def authorize!
+    authenticate_user!
+
+    return current_user if current_user.id.to_s == params[:user_id]
+
+    redirect_to users_path, alert: I18n.t('denial.forbidden')
+    nil
```

Comment:
```suggestion
    return redirect_to users_path, alert: I18n.t('denial.forbidden') unless current_user.id.to_s == params[:user_id]
```

Reply:
Ah I see why you had to change this. I think it would be possible to use `return if authorize!` after my change because it only returns a value when redirecting but then the method name wouldn't be very intuitive. But because that's also not very pretty it's ok to leave it like it is now :)